### PR TITLE
feat: add offline-installation-tool.sh

### LINKED ISSUE #109: kosmosctl command should support offline installation
**What would you like to be added**:
kosmosctl command can support offline installation

**Why is this needed**:
`kosmosctl image pull --image-list=[image-list.txt] --output=[output-dir] --container-runtime=[container-runtime]`
`kosmoscrl image push --image-list=[image-list.txt] --artifact=[*.tar.gz] --username=[registry-username] --password=[registry-password] --private-registry=[private-registry-name]
`

### PR DESCRIPTION
**Pull request by kuangcheng66**
#### What type of PR is this?
- **feat:** offline-installation-tool

#### What does this PR do?
1. docker pull/save images in images-list and save images in image-dir
2. docekr load/tag/puhs image on your local machine and  push it to the private-registry

#### Which issue(s) does this PR fix?
Fixes #109

### CODE REVIEW

File: hack/tools/offline-installation-tool.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The KubeSphere Authors.
```
Comment (ONE7live) [inline hack/tools/offline-installation-tool.sh]: Certificate information needs attention

Comment (kuangcheng66) [reply hack/tools/offline-installation-tool.sh]: ok, I will fix it

File: hack/tools/offline-installation-tool.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    # count the number of images in ImageList
+    ImagesListLen=$(cat ${ImagesList} | wc -l)
+    name=""
+    images=""
+    index=0
+    for image in $(<${ImagesList}); do
+        if [[ ${image} =~ ^\#\#.* ]]; then
+           if [[ -n ${images} ]]; then
+              echo ""
```
Comment (ONE7live) [inline hack/tools/offline-installation-tool.sh]: change echo "" —> echo -e "a\b\..." maybe better, what do you think?

File: hack/tools/offline-installation-tool.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+       fi
+       # docker retag and push images
+       ImageName=${image##*/}
+       imageUrl=$registryUrl"/"$ImageName
+       image=$(echo "${image}" |tr -d '\r')
+       imageUrl=$(echo "${imageUrl}" |tr -d '\r')
+       docker tag $image $imageUrl
+       docker push $imageUrl
+     done
+fi
```
Comment (duanmengkk) [inline hack/tools/offline-installation-tool.sh]: add a new line

**Comment by duanmengkk** [comment]
This shell script doesn't seem to be an 'offline installation tool,' but rather appears to be an image handling script？

File: hack/tools/offline-installation-tool.sh (hunk 2 above)
Comment (kuangcheng66) [reply hack/tools/offline-installation-tool.sh]: ok

File: hack/tools/offline-installation-tool.sh (hunk 3 above)
Comment (kuangcheng66) [reply hack/tools/offline-installation-tool.sh]: ok

**Comment by kuangcheng66** [comment]
> This shell script doesn't seem to be an 'offline installation tool,' but rather appears to be an image handling script？

yes，it seems that we still need to discuss it carefully

**Pull request closed without merging**